### PR TITLE
Deploy dex on minikube

### PR DESCRIPTION
**Pull request by tolusha**
### What does this PR do?
Deploy dex on minikube

### Screenshot/screencast of this PR

```
  ✔ Deploy Dex
    ✔ Create Dex namespace...[OK]
    ✔ Create Dex certificate
      ✔ Check Cert Manager deployment...not deployed
      ✔ Deploy Cert Manager...done
      ✔ Wait for Cert Manager...ready
      ✔ Check Cert Manager CA certificate...generating new one
      ✔ Set up Eclipse Che certificates issuer...done
      ✔ Request self-signed certificate...done
      ✔ Wait for self-signed certificate...ready
      ✔ Save Dex certificate...[OK: /tmp/dex-ca.crt]
    ✔ Create Dex service account...[OK]
    ✔ Create Dex cluster role...[OK]
    ✔ Create Dex cluster role binding...[OK]
    ✔ Create Dex service...[OK]
    ✔ Create Dex configmap...[OK]
    ✔ Create Dex deployment...[OK]
    ✔ Wait for Dex is ready...[OK]
    ✔ Configure API server
      ✔ Configure Minikube API server...[OK]
      ✔ Wait for Minikube API server...[OK]
```      

### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/19366

### How to test this PR?

```bash
# deploy dex
cat >/tmp/patch.yaml <<EOF                                                          
spec:
  auth:
    nativeUserMode: true
EOF

yarn && cd bin && ./run server:deploy --platform minikube --che-operator-cr-patch-yaml /tmp/patch.yaml

# check dex pod logs for errors
DEX_POD=$(kubectl get pods -n dex | grep dex | awk '{print $1}')
kubectl logs $DEX_POD -n dex
```

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [ ] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

**Comment by sparkoo** [comment]
I can't yarn the branch, is it issue on my side? Sorry if it is obvious issue, I don't work with yarn or javascript projects very often...

```
[~/dev/chectl] (dex ✘)✭ λ yarn
yarn install v1.22.4
[1/5] Validating package.json...
[2/5] Resolving packages...
[3/5] Fetching packages...
error Couldn't find match for "b8cc02bf9a6a0d6f4eb5d64e02feae8b51653bcd" in "refs/heads/1.x,refs/heads/19284,refs/heads/19978,refs/heads/20337,refs/heads/6.18.x,refs/heads/7.10.0,refs/heads/7.12.0,refs/heads/7.12.1,refs/heads/7.12.2,refs/heads/7.12.x,refs/heads/7.13.0,refs/heads/7.13.1,refs/heads/7.13.x,refs/heads/7.14.x,refs/heads/7.15.x,refs/heads/7.16.x,refs/heads/7.17.x,refs/heads/7.18.x,refs/heads/7.19.x,refs/heads/7.20.x,refs/heads/7.21.x,refs/heads/7.22.x,refs/heads/7.23.x,refs/heads/7.24.x,refs/heads/7.25.x,refs/heads/7.26.x,refs/heads/7.27.x,refs/heads/7.28.x,refs/heads/7.29.x,refs/heads/7.3.0,refs/heads/7.3.1,refs/heads/7.3.2,refs/heads/7.30.x,refs/heads/7.31.x,refs/heads/7.32.x,refs/heads/7.33.x,refs/heads/7.34.x,refs/heads/7.35.x,refs/heads/7.36.x,refs/heads/7.37.x,refs/heads/7.4.0,refs/heads/7.6.0,refs/heads/7.7.0,refs/heads/7.8.0,refs/heads/7.9.1,refs/heads/7.9.2,refs/heads/7.9.3,refs/heads/7.9.x,refs/heads/7.test,refs/heads/AddWorkspaceLogsToTheCIChecks,refs/heads/che-17090,refs/heads/che-19837,refs/heads/che-20200,refs/heads/che-20547,refs/heads/che-ci,refs/heads/cheos,refs/heads/crw-2.0,refs/heads/defaultProtocol,refs/heads/disc_back,refs/heads/dwo080-released-main,refs/heads/fix-18393,refs/heads/fix-pr-gh,refs/heads/fixCheOperatorArchCompilation,refs/heads/fixUsingCustomImageWithDeployDevCommand,refs/heads/fix_disc,refs/heads/fix_mem,refs/heads/fixes_ci,refs/heads/fixxxa,refs/heads/gh-release,refs/heads/improvetest,refs/heads/keycloak.11.0.2.Test,refs/heads/keycloakInternalNetwork,refs/heads/main,refs/heads/managedDWO,refs/heads/newerOperatorSDK-before-squash,refs/heads/ocp_ci,refs/heads/openshiftci,refs/heads/operator-tls,refs/heads/operatorPRgenFix,refs/heads/opmBundle,refs/heads/opmBundle2,refs/heads/opt_tel,refs/heads/platforms-pr,refs/heads/pr-update-base-images-1627071868,refs/heads/pr-update-base-images-1627073652,refs/heads/pr-update-base-images-1628900332,refs/heads/pr-update-base-images-1629035233,refs/heads/refactor-dir-variables,refs/heads/release-fix,refs/heads/reuseMakeFileTemplateGenerationInTheCITests,refs/heads/test_ci,refs/heads/test_release,refs/tags/1.0.1.GA,refs/tags/1.0.2.GA,refs/tags/1.1.0.GA,refs/tags/1.2.0.GA,refs/tags/1.2.2.GA,refs/tags/2.0.0.GA,refs/tags/2.1.0.GA,refs/tags/2.1.1.GA,refs/tags/2.2.0.GA,refs/tags/2.3.0.GA,refs/tags/6.19.2,refs/tags/7.0.0,refs/tags/7.0.0-RC-2.0,refs/tags/7.0.0-rc-4.0,refs/tags/7.1.0,refs/tags/7.10.0,refs/tags/7.11.0,refs/tags/7.12.0,refs/tags/7.12.1,refs/tags/7.19.0,refs/tags/7.19.1,refs/tags/7.19.2,refs/tags/7.2.0,refs/tags/7.20.0,refs/tags/7.20.1,refs/tags/7.21.0,refs/tags/7.21.1,refs/tags/7.21.2,refs/tags/7.22.0,refs/tags/7.22.1,refs/tags/7.22.2,refs/tags/7.23.0,refs/tags/7.23.1,refs/tags/7.23.2,refs/tags/7.24.0,refs/tags/7.24.1,refs/tags/7.24.2,refs/tags/7.25.0,refs/tags/7.25.1,refs/tags/7.25.2,refs/tags/7.26.0,refs/tags/7.26.1,refs/tags/7.26.2,refs/tags/7.27.0,refs/tags/7.27.1,refs/tags/7.27.2,refs/tags/7.28.0,refs/tags/7.28.1,refs/tags/7.28.2,refs/tags/7.29.0,refs/tags/7.29.1,refs/tags/7.29.2,refs/tags/7.3.0,refs/tags/7.3.1,refs/tags/7.3.2,refs/tags/7.30.0,refs/tags/7.30.1,refs/tags/7.30.2,refs/tags/7.31.0,refs/tags/7.31.1,refs/tags/7.31.2,refs/tags/7.32.0,refs/tags/7.32.1,refs/tags/7.32.2,refs/tags/7.33.0,refs/tags/7.33.1,refs/tags/7.33.2,refs/tags/7.34.0,refs/tags/7.34.1,refs/tags/7.34.2,refs/tags/7.35.0,refs/tags/7.35.1,refs/tags/7.35.2,refs/tags/7.36.0,refs/tags/7.36.1,refs/tags/7.36.2,refs/tags/7.37.0,refs/tags/7.4.0,refs/tags/7.6.0,refs/tags/7.7.0,refs/tags/7.7.1,refs/tags/7.8.0,refs/tags/7.9.0,refs/tags/7.9.1,refs/tags/7.9.2,refs/tags/7.9.3,refs/tags/v7.12.2,refs/tags/v7.13.0,refs/tags/v7.13.1,refs/tags/v7.13.2,refs/tags/v7.14.1,refs/tags/v7.14.2,refs/tags/v7.14.3,refs/tags/v7.15.0,refs/tags/v7.15.1,refs/tags/v7.15.2,refs/tags/v7.16.0,refs/tags/v7.16.1,refs/tags/v7.16.2,refs/tags/v7.17.0,refs/tags/v7.17.1,refs/tags/v7.17.2,refs/tags/v7.18.0,refs/tags/v7.18.1,refs/tags/v7.18.2" for "git://github.com/eclipse-che/che-operator".
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
```

**Comment by tolusha** [comment]
@sparkoo 
try `yarn cache clean`

File: src/tasks/installers/helm.ts [hunk 1]
```diff
@@ -117,7 +117,13 @@ export class HelmTasks {
             const certManagerTasks = new CertManagerTasks(flags)
             const certManagerListTasks = new Listr(undefined, ctx.listrOptions)
             certManagerListTasks.add(certManagerTasks.getDeployCertManagerTasks(flags))
-            certManagerListTasks.add(certManagerTasks.getGenerateCertificatesTasks(flags))
+            certManagerListTasks.add(certManagerTasks.getGenerateCertManagerCACertificateTasks(flags))
```
Comment (mmorhun) [inline src/tasks/installers/helm.ts:120]:
```suggestion
            certManagerTasksList.add(certManagerTasks.getGenerateCertManagerCACertificateTasks(flags))
```

File: src/tasks/platforms/minikube.ts [hunk 2]
```diff
@@ -131,6 +131,50 @@ export class MinikubeTasks {
     ], { renderer: flags['listr-renderer'] as any })
   }
 
+  configureApiServer(flags: any): ReadonlyArray<Listr.ListrTask> {
+    return [
+      {
+        title: 'Copy certificate into Minikube',
```
Comment (mmorhun) [inline src/tasks/platforms/minikube.ts:137]:
```suggestion
        title: 'Copy DEX certificate into Minikube',
```

File: src/tasks/platforms/minikube.ts [hunk 3]
```diff
@@ -131,6 +131,50 @@ export class MinikubeTasks {
...
+          task.title = `${task.title}...[OK]`
+        },
+      },
+      {
+        title: 'Configure Minikube API server',
+        task: async (ctx: any, task: any) => {
+          const args: string[] = []
+          args.push(`--extra-config=apiserver.oidc-issuer-url=${ctx[OIDCContextKeys.ISSUER_URL]}`)
+          args.push(`--extra-config=apiserver.oidc-client-id=${ctx[OIDCContextKeys.CLIENT_ID]}`)
+          args.push('--extra-config=apiserver.oidc-ca-file=/etc/ca-certificates/dex-ca.crt')
```
Comment (mmorhun) [inline src/tasks/platforms/minikube.ts:155]: Do we need to add this certificate to Che trusted CA bundle?

File: yarn.lock [hunk 4]
```diff
@@ -2645,7 +2645,7 @@ ecc-jsbn@~0.1.1:
 
 "eclipse-che-operator@git://github.com/eclipse-che/che-operator#main":
   version "0.0.0"
-  resolved "git://github.com/eclipse-che/che-operator#d52d5adb16e751699f257561836b3582cd6e26f9"
+  resolved "git://github.com/eclipse-che/che-operator#b8cc02bf9a6a0d6f4eb5d64e02feae8b51653bcd"
```
Comment (mmorhun) [inline yarn.lock:2648]: I think you don't relay on operator resources in this PR, so this change will only cause merge conflicts.

File: src/tasks/platforms/platform.ts [hunk 5]
```diff
@@ -22,18 +22,33 @@ import { OpenshiftTasks } from './openshift'
...
+  protected dockerDesktopTasks: DockerDesktopTasks
 
+  constructor(flags: any) {
+    this.minikubeTasks = new MinikubeTasks()
+    this.microk8sTasks = new MicroK8sTasks()
+    this.minishiftTasks = new MinishiftTasks()
+    this.openshiftTasks = new OpenshiftTasks()
+    this.k8sTasks = new K8sTasks()
+    this.crc = new CRCHelper()
+    this.dockerDesktopTasks = new DockerDesktopTasks(flags)
```
Comment (mmorhun) [inline src/tasks/platforms/platform.ts:48]: Why do we need to create all these tasks if we going to use only one list?

**Comment by sparkoo** [comment]
Fails to deploy Dex for me.

```
    ✔ Create Dex deployment...[OK]
    ✖ Wait for Dex is ready
      → ERR_TIMEOUT: Timeout set to pod ready timeout 600000
      Configure API server
    🏃‍  Running the Eclipse Che operator
    Error: Command server:deploy failed. Error log: /home/mvala/.cache/chectl/error.log.
```

```
[~/dev/chectl/bin] (dex ✘)✹✭ λ kc get pods -n dex
NAME                 READY   STATUS             RESTARTS   AGE
dex-d79c6f7d-pltwq   0/1     CrashLoopBackOff   7          12m
```

```
[~/dev/chectl/bin] (dex ✘)✹✭ λ kc logs dex-d79c6f7d-pltwq -n dex
time="2021-10-04T11:47:09Z" level=info msg="config issuer: https://dex.192.168.39.130.nip.io:32000"
failed to initialize storage: failed to inspect service account token: jwt claim "kubernetes.io/serviceaccount/namespace" not found
```

**Comment by tolusha** [comment]
@sparkoo 
What is your minikube version?

**Comment by sparkoo** [comment]
> @sparkoo What is your minikube version?

```
[~/dev/chectl] (dex ✘)✹✭ λ minikube version
minikube version: v1.23.0
commit: 5931455374810b1bbeb222a9713ae2c756daee10
```

```
[~/dev/chectl] (dex ✘)✹✭ λ kubectl version
Client Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.2", GitCommit:"8b5a19147530eaac9476b0ab82980b4088bbc1b2", GitTreeState:"clean", BuildDate:"2021-09-15T21:38:50Z", GoVersion:"go1.16.8", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.1", GitCommit:"632ed300f2c34f6d6d15ca4cef3d3c7073412212", GitTreeState:"clean", BuildDate:"2021-08-19T15:39:34Z", GoVersion:"go1.16.7", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by tolusha** [comment]
I tested on minikube v1.21.0 and reproduced issue with v1.23.0

File: src/tasks/component-installers/dex.ts [hunk 6]
```diff
@@ -0,0 +1,237 @@
...
+import { CertManagerTasks } from './cert-manager'
+
+namespace Template {
+  export const DOMAIN = '{{DOMAIN}}'
+  export const CHE_NAMESPACE = '{{NAMESPACE}}'
+  export const CLIENT_ID = '{{CLIENT_ID}}'
+  export const CLIENT_SECRET = '{{CLIENT_SECRET}}'
+}
+
+export class DexTasks {
```
[the remark this answers is not in the document]
Comment (tolusha) [reply src/tasks/component-installers/dex.ts:36]: Helm is out of scope.

File: src/tasks/component-installers/dex.ts [hunk 7]
```diff
@@ -0,0 +1,237 @@
...
+                }])
+
+                return certs
+              },
+            },
+            {
+              title: 'Create Dex service account',
+              task: async (_: any, task: any) => {
+                if (await this.kube.isServiceAccountExist(this.dexName, this.namespaceName)) {
+                  task.title = `${task.title}...[Exists]`
```
[the remark this answers is not in the document]
Comment (tolusha) [reply src/tasks/component-installers/dex.ts:119]: Basically about same

File: src/tasks/component-installers/dex.ts [hunk 8]
```diff
@@ -0,0 +1,237 @@
...
+              task: async (_: any, task: any) => {
+                await this.kube.waitForPodReady(this.selector, this.namespaceName)
+                task.title = `${task.title}...[OK]`
+              },
+            },
+            {
+              title: 'Configure API server',
+              task: async (ctx: any) => {
+                ctx[OIDCContextKeys.CLIENT_ID] = this.clientId
+                ctx[OIDCContextKeys.ISSUER_URL] = `https://dex.${this.flags.domain}:32000`
```
[the remark this answers is not in the document]
Comment (tolusha) [reply src/tasks/component-installers/dex.ts:227]: What do you mean?

**Comment by tolusha** [comment]
To be honest initial scripts [1] don't work with minikube v1.23 as well
[1] https://github.com/che-incubator/che-auth-playground/tree/master/minikube_dex

**Comment by sparkoo** [comment]
@tolusha ok. I've initially run with older kubernetes version on latest minikube `--kubernetes-version='1.21.4'`, but that doesn't help. I'll try older minikube then, but I guess we should fix that.

**Comment by sparkoo** [comment]
works with minikube 1.21

**Comment by sparkoo** [comment]
I believe you're using different self-signed certificates for Dex and Che. Does it deploy CA of Dex self signed cert anywhere in eclipse-che namespace? I believe we need that in order to communicate with Dex over https from Che.

File: src/tasks/platforms/platform.ts (hunk 5 above)
Comment (tolusha) [reply src/tasks/platforms/platform.ts:48]: This tasks are moved to constructor not to duplicate in every function.

File: src/tasks/platforms/minikube.ts (hunk 3 above)
Comment (tolusha) [reply src/tasks/platforms/minikube.ts:155]: @sparkoo mentioned that it is needed. I will rework.

File: installers/dex/configmap.yaml [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+    - id: {{CLIENT_ID}}
+      redirectURIs:
+      - 'https://{{NAMESPACE}}.{{DOMAIN}}/oauth2/callback'
+      name: 'Eclipse Che'
+      secret: {{CLIENT_SECRET}}
+    enablePasswordDB: true
+    staticPasswords:
+    - email: "che@eclipse.org"
+      # bcrypt hash of the string "password"
+      hash: "$2a$10$2b2cU8CPhOTaGrs1HRQuAueS7JTT5ZHsHSzYiFPm1leZck7Mc8T4W"
```
[the remark this answers is not in the document]
Comment (tolusha) [reply installers/dex/configmap.yaml:33]: @sparkoo 
Could you answer pls?

File: installers/dex/cluster-role.yaml [hunk 10]
```diff
@@ -0,0 +1,14 @@
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
```
[the remark this answers is not in the document]
Comment (tolusha) [reply installers/dex/cluster-role.yaml:3]: It is related minikube configuration not specific Eclipse Che version

File: installers/dex/service.yaml [hunk 11]
```diff
@@ -0,0 +1,18 @@
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: dex
+  namespace: dex
+  labels:
+    app: dex
+spec:
+  type: NodePort
```
[the remark this answers is not in the document]
Comment (tolusha) [reply installers/dex/service.yaml:10]: That's a sample taken from Dex

File: src/commands/server/deploy.ts [hunk 12]
```diff
@@ -234,8 +235,8 @@ export default class Deploy extends Command {
    * Checks if TLS is disabled via operator custom resource.
    * Returns true if TLS is enabled (or omitted) and false if it is explicitly disabled.
    */
-  async checkTlsMode(ctx: any): Promise<boolean> {
-    const crPatch = ctx.crPatch
+  private getTlsSupport(ctx: any): boolean {
```
[the remark this answers is not in the document]
Comment (tolusha) [reply src/commands/server/deploy.ts:238]: I would not say support but codebase still exists.

File: installers/dex/configmap.yaml (hunk 9 above)
[the remark this answers is not in the document]
Comment (sparkoo) [reply installers/dex/configmap.yaml:33]: I don't think it's needed. It's never meant to be used in production. It's like the admin:admin we now have in keycloak...

File: installers/dex/service.yaml (hunk 11 above)
[the remark this answers is not in the document]
Comment (sparkoo) [reply installers/dex/service.yaml:10]: I'm not sure as well. I believe it's needed for Kubernetes apiserver and at the same time it has to be publicly available for oauth2 process flow. If you make it work with something better, I'm all in.

**Comment by tolusha** [comment]
@sparkoo 
ready to review again, pls use minikube v1.23.2

@mmorhun 
addressed all remarks but using cluster default certificates (in progress) & ingress

File: installers/dex/configmap.yaml (hunk 9 above)
[the remark this answers is not in the document]
Comment (mmorhun) [reply installers/dex/configmap.yaml]: AFAIK, we don't have `admin:admin` in Keycloak for considerable amount of time. All the passwords are generated and stored in secrets.

File: src/tasks/platforms/platform.ts (hunk 5 above)
Comment (mmorhun) [reply src/tasks/platforms/platform.ts:48]: It changed nothing for now, but ok. The refactoring is probably out of scope of this PR

**Comment by sparkoo** [comment]
I'm getting this:
```
      ✖ Add Dex certificate to Eclipse Che certificates bundle
        → namespaces "eclipse-che" not found
```

**Comment by tolusha** [comment]
> namespaces "eclipse-che" not found

fixed

File: src/tasks/component-installers/dex.ts [hunk 13]
```diff
@@ -0,0 +1,326 @@
...
+                    if (await this.kube.isConfigMapExists(DexCaConfigMap.NAME, this.flags.chenamespace)) {
+                      task.title = `${task.title}...[Exists]`
+                    } else {
+                      const dexCa = new V1ConfigMap()
+                      dexCa.metadata = new V1ObjectMeta()
+                      dexCa.metadata.name = DexCaConfigMap.NAME
+                      dexCa.metadata.labels = DexCaConfigMap.LABELS
+                      dexCa.data = { 'ca.crt': ctx[DexContextKeys.DEX_CA_CRT] }
+
+                      await this.kube.createNamespacedConfigMap(this.flags.chenamespace, dexCa)
```
Comment (mmorhun) [inline src/tasks/component-installers/dex.ts:148]: Where is the labels for marking a config map as a trusted CAs bundle config map added?

File: src/tasks/component-installers/dex.ts [hunk 14]
```diff
@@ -0,0 +1,326 @@
...
+                }
+              },
+            },
+            {
+              title: 'Create Dex configmap',
+              task: async (ctx: any, task: any) => {
+                const dexConfigMap = await this.kube.getConfigMap(this.dexName, this.namespaceName)
+                if (dexConfigMap && dexConfigMap.data) {
+                  // read client secret
+                  const config = yaml.load(dexConfigMap.data['config.yaml']) as any
```
Comment (mmorhun) [inline src/tasks/component-installers/dex.ts:248]: Are you sure, that `yaml.load` tolerate `undefined` as an argument?

File: src/tasks/platforms/minikube.ts [hunk 15]
```diff
@@ -131,6 +131,55 @@ export class MinikubeTasks {
     ], { renderer: flags['listr-renderer'] as any })
   }
 
+  configureApiServer(flags: any): ReadonlyArray<Listr.ListrTask> {
```
Comment (mmorhun) [inline src/tasks/platforms/minikube.ts:134]: Please give more specific name for the function, for example `configureApiServerForDex`.

File: src/tasks/platforms/minikube.ts [hunk 16]
```diff
@@ -131,6 +131,55 @@ export class MinikubeTasks {
...
+          args.push('start')
+
+          await execa('minikube', args, { timeout: 60000 })
+
+          task.title = `${task.title}...[OK]`
+        },
+      },
+      {
+        title: 'Wait for Minikube API server',
+        task: async (_: any, task: any) => {
```
Comment (mmorhun) [inline src/tasks/platforms/minikube.ts:172]:
```suggestion
        task: async (_ctx: any, task: any) => {
```

File: src/tasks/platforms/platform.ts [hunk 17]
```diff
@@ -90,4 +105,12 @@ export class PlatformTasks {
 
     return [task]
   }
+
+  configureApiServer(flags: any): ReadonlyArray<Listr.ListrTask> {
```
Comment (mmorhun) [inline src/tasks/platforms/platform.ts:109]: Please be more specific in the name, or at least add a comment

**Comment by sparkoo** [comment]
Client will need client-id and secret. I assume that client-id is static `eclipse-che`. What about secret? Is it available anywhere in `eclipse-che` namespace?

Btw, there is also way how to configure dex programatically https://dexidp.io/docs/api/#go. So in theory, we can configure the client from che-operator. However, I've never tried that and I don't think it's worth the effort. Static configuration is imho fine for now.

File: package.json [hunk 18]
```diff
@@ -94,7 +96,7 @@
   "files": [
     "/bin",
     "/lib",
-    "/installers",
+    "/resources",
```
[the remark this answers is not in the document]
Comment (tolusha) [reply package.json:99]: Agree

File: src/tasks/component-installers/dex.ts (hunk 14 above)
Comment (tolusha) [reply src/tasks/component-installers/dex.ts:248]: Agree. Will fix.

File: resources/dex/configmap.yaml [hunk 19]
```diff
@@ -0,0 +1,34 @@
...
+    staticClients:
+    - id: {{CLIENT_ID}}
+      redirectURIs:
+      - 'https://{{NAMESPACE}}.{{DOMAIN}}/oauth2/callback'
+      name: 'Eclipse Che'
+      secret: {{CLIENT_SECRET}}
+    enablePasswordDB: true
+    staticPasswords:
+    - email: "che@eclipse.org"
+      hash: "{{DEX_PASSWORD_HASH}}"
```
[the remark this answers is not in the document]
Comment (tolusha) [reply resources/dex/configmap.yaml:32]: Ok WIll code for a while but set fixed `admin:admin` credentials for now

File: src/tasks/component-installers/dex.ts (hunk 13 above)
Comment (mmorhun) [reply src/tasks/component-installers/dex.ts:154]: Sorry, missed it. Thanks for pointing out.

File: resources/dex/configmap.yaml [hunk 20]
```diff
@@ -0,0 +1,34 @@
...
+      tlsCert: /etc/dex/tls/tls.crt
+      tlsKey: /etc/dex/tls/tls.key
+
+    oauth2:
+      skipApprovalScreen: true
+
+    staticClients:
+    - id: {{CLIENT_ID}}
+      redirectURIs:
+      - 'https://{{NAMESPACE}}.{{DOMAIN}}/oauth2/callback'
```
Comment (sparkoo) [inline resources/dex/configmap.yaml:26]: this is not where che is exposed
```
    ✔ Users Dashboard           : https://che-eclipse-che.192.168.39.106.nip.io
```

File: resources/dex/configmap.yaml [hunk 21]
```diff
@@ -0,0 +1,34 @@
...
+    app: dex
+data:
+  config.yaml: |
+    issuer: https://dex.{{DOMAIN}}
+    storage:
+      type: kubernetes
+      config:
+        inCluster: true
+    web:
+      https: 0.0.0.0:5556
```
Comment (sparkoo) [inline resources/dex/configmap.yaml:16]: I think we need to listen on http as tls is already manager in ingress

**Comment by sparkoo** [comment]
@tolusha thank you. Works for me now as expected.

**Pull request merged**